### ISSUE TITLE
Info: bloc_test 8.0.1 introduced breaking changes compared to 8.0.0 when importing mockito

### ISSUE THREAD

**Issue by b3arco** [felangel/bloc#2510]
**Describe the bug**
`bloc_test` version 8.0.1 introduces a breaking change in specific setups compared to version 8.0.0.
See PR #2473 for the update.
In particular "feat: `export mocktail`" causes this if in in a test class, the package `mockito` was also imported and `mocktail` was not.

I realize it is was probably unintended in version 8.0.0 to begin with that it was possible to import both `mockito` and `bloc_test` and use them side-by-side. However, this did work just fine.

I had test code that was working as expected.
When I now update to `bloc_test` 8.0.1, I get syntax errors and my code fails to compile.

**To Reproduce**
1. Have a test class that uses `bloc_test` 8.0.0 for bloc tests and `mockito` for mocks in that class.
2. Update `bloc_test` to 8.0.1
4. See syntax errors

**Screenshots**
<img width="1268" alt="Testing code that imports bloc_test and mockito. It defines a class Mock Client that extends Mock and implements Client. The word Mock is marked with an error" src="https://user-images.githubusercontent.com/81265357/120530226-929ec700-c3dd-11eb-99c4-67a93af3a7e8.png">

**Additional context**
I don't want to blame anyone. The update is already released and I can imagine that this was originally planned for the 8.0.0 release, so I guess it's fine. I just wanted to bring this to your attention.

**Comment by felangel** [comment]
Hi @b3arco 👋
Thanks for opening an issue and sorry for the inconvenience! This was planned for the v8.0.0 release as you mentioned and you can still use both by hiding Mock from bloc_test:

```dart
import ‘package:bloc_test/bloc_test.dart’ hide Mock;
```

Again sorry for an inconvenience!

**Comment by yerzhant** [comment]
Hi @felangel 

First of all thanks for great tools.

I've faced almost the same issue but with `when`. Not to write `hide when` on many places just reverted back to 8.0.0. Wouldn't it be more appropriate/convenient (from usage perspective) to exclude `export 'package:mocktail/mocktail.dart';` from `bloc_test.dart` and an end user will import mocktail if there's necessity for it? Like this:

```
...
import 'package:mockito/mockito.dart';
import 'package:mocktail/mocktail.dart' as mocktail;
...
  setUpAll(() {
    mocktail.registerFallbackValue(const FavoriteState([]));
    mocktail.registerFallbackValue(const FavoriteEvent.show());
  });
...
    blocTest(
      'should load chats',
      build: () {
        when(repo.getMy(0, 20)).thenAnswer(
...
```

**Comment by felangel** [comment]
@yerzhant @b3arco yeah I thought about it some more and I agree it's probably better to just revert the change in order to eliminate conflicts with `mockito`.

#2513 reverts the export and it should be fixed in v8.0.2. Sorry again for the inconvenience!
